### PR TITLE
Bring CI up to latest and greatest, drop upper bounds

### PR DESCRIPTION
**Pull request by mjgpy3**
This is following the patterns of https://github.com/freckle/faktory_worker_haskell/pull/59

The only way this really differs it that it makes the CHANGELOG update too.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
     steps:
       - uses: actions/checkout@v2
-      - uses: freckle/stack-upload-action@main
+
+      - id: tag
+        uses: freckle/haskell-tag-action@main
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - if: steps.tag.tag
```
Comment (pbrisbin) [inline .github/workflows/release.yml]: I had a bug in the reference PR, which I commented on after the fact. Oops.

```suggestion
      - if: steps.tag.outputs.tag
```

**Review by pbrisbin — APPROVED** [review]
I had been keeping the upper bound on `base`, but I don't know why now. YOLO.

File: .github/workflows/release.yml (hunk 1 above)
Comment (mjgpy3) [reply .github/workflows/release.yml]: Thanks!

**Comment by mjgpy3** [comment]
@pbrisbin from the "Release" action
> The dependency 'build-depends: base' does not specify an upper bound on the version number. Each major release of the 'base' package changes the API in various ways and most packages will need some changes to compile with it. The recommended practise is to specify an upper bound on the version of the 'base' package. This ensures your package will continue to build when a new major version of the 'base' package is released. If you are not sure what upper bound to use then use the next  major version. For example if you have tested your package with 'base' version 4.5 and 4.6 then use 'build-depends: base >= 4.5 && < 4.7'.


I guess that's why :rofl: :sob:

**Comment by pbrisbin** [comment]
Ha, good ol Hackage checks. I'm pleasantly surprised the API error from the upload came back through to you so clearly. I would've totally expected `Error (400)` or something; go me.

**Pull request merged**